mage/releases/publish.go

```diff
@@ -217,3 +231,35 @@ func listFiles(dir string) []string {
 
 	return names
 }
+
+func generateChecksumFile(path string) (string, error) {
+	if filepath.Ext(path) == "sha256sum" {
+		return path, nil
+	}
+	data, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer data.Close()
+
+	f, err := os.Create(filepath.Join(path + ".sha256sum"))
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	hash := sha256.New()
+	if _, err := io.Copy(hash, data); err != nil {
+		return "", err
+	}
+	sum := hash.Sum(nil)
+
+	// write the checksum and file name to the checksum file so it can be
+	// verified by tools like `shasum`
+	_, err = f.WriteString(fmt.Sprintf("%x  %s", sum[:], filepath.Base(path)))
+	if err != nil {
+		return "", err
```

Comment:
Even though this is just our build script, it's still super helpful to wrap the errors from low level calls like what's in this function with `errors.Wrap` and provide context into what we were trying to do when it failed. It helps avoid a build failing with "EOF" or "file not found" without any clue what we were trying to do. 😁

Reply:
I broke down the function into more modular pieces. Let me know if that helps